### PR TITLE
[doc] Fix rstmgr documentation

### LINKED ISSUE #15982: [*mgr] Align documentation with reset domain reorg
As title, to make sure the docs are consistent with https://github.com/lowRISC/opentitan/issues/15317.
Can be done post M2.

### PR DESCRIPTION
**Pull request by tjaychen**
- update for latest reset organization
- fixes #15982

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

File: hw/ip/rstmgr/doc/_index.md [hunk 1]
```diff
@@ -170,24 +156,20 @@ The reset manager then checks as follows:
...
 
-{{< incGenFromIpDesc "/hw/top_earlgrey/ip/rstmgr/data/autogen/rstmgr.hjson" "hwcfg" >}}
+### Parameters
+
+The following table lists the instantiation parameters of `rstmgr`.
+
+
+Parameter                   | Default       | Description
+----------------------------|---------------|---------------
+`SecCheck`                  | 1             | Enables reset consistency checks on the leaf reset.  Each check contains a small FSM
```
Comment (msfschaffner) [inline hw/ip/rstmgr/doc/_index.md:166]: nitnit: missing period at the end of the sentence

File: hw/ip/rstmgr/doc/_index.md [hunk 2]
```diff
@@ -170,24 +156,20 @@ The reset manager then checks as follows:
...
-`cpu_dump_i`            | `input`   | CPU crash dump state from `rv_core_ibex`.
-`pwr_i.rst_lc_req`      | `input`   | Power manager request to assert the `rst_lc_n` tree.
-`pwr_i.rst_sys_req`     | `input`   | Power manager request to assert the `rst_sys_n` tree.
-`pwr_i.reset_cause`     | `input`   | Power manager indication for why it requested reset, the cause can be low power entry or peripheral issued request.
-`pwr_i.rstreqs`         | `input`   | Peripheral reset requests.
-`pwr_o.rst_lc_src_n`    | `output`  | Current state of `rst_lc_n` tree.
-`pwr_o.rst_sys_src_n`   | `output`  | Current state of `rst_sys_n` tree.
-`resets_ast_o`          | `output`  | Resets used by `ast`.
-`resets_o`              | `output`  | Resets used by the rest of the core domain.
+{{< incGenFromIpDesc "/hw/top_earlgrey/ip/rstmgr/data/autogen/rstmgr.hjson" "hwcfg" >}}
```
Comment (msfschaffner) [inline hw/ip/rstmgr/doc/_index.md:172]: If you do not want to loose the signal descriptions here, you can transfer them to the `desc` key of the corresponding signals in the hjson file. I.e., the docgen script takes the desc from there and populates the table with that information.

**Review by msfschaffner — APPROVED** [review]

File: hw/ip/rstmgr/doc/_index.md (hunk 2 above)
Comment (tjaychen) [reply hw/ip/rstmgr/doc/_index.md:172]: o yes, thanks for the reminder. I was actually going to beef up the hjson description but completely forgot.

Comment (tjaychen) [reply hw/ip/rstmgr/doc/_index.md:172]: updated.

File: hw/ip/rstmgr/doc/_index.md (hunk 1 above)
Comment (tjaychen) [reply hw/ip/rstmgr/doc/_index.md]: fixed.

File: hw/ip/rstmgr/data/rstmgr.hjson.tpl [hunk 3]
```diff
@@ -135,47 +135,70 @@
       name:    "por_n",
       act:     "rcv",
       width:   "${len(power_domains)}"
+      desc:    '''
+        Root power on reset signals from ast.
+        There is one root reset siganl for each core power domain.
```
Comment (msfschaffner) [inline hw/ip/rstmgr/data/rstmgr.hjson.tpl:140]: nit: siganl

**Review by msfschaffner — APPROVED** [review]
Another nit, otherwise LGTM

File: hw/ip/rstmgr/data/rstmgr.hjson.tpl (hunk 3 above)
Comment (tjaychen) [reply hw/ip/rstmgr/data/rstmgr.hjson.tpl]: ugh, sorry for the typo, not sure why my spell check didn't flag this...

**Comment by jeoongp** [comment]
@tjaychen sorry for late reply and thanks for updating doc. 
Updated doc and summary are very helpful!

**Pull request merged**